### PR TITLE
TuneD documentation

### PR DESCRIPTION
**Pull request by sounix000**
### PR creator: Description

Provide decent documentation on the [TuneD](https://tuned-project.org) application and associated utilities, in the context of SLES. The requirement for documentation is described in the [Bugzilla#1093289](https://bugzilla.suse.com/show_bug.cgi?id=1093289). The [source code](https://github.com/redhat-performance/tuned) for the application is available on GitHub.

### PR creator: Are there any relevant issues/feature requests?

* [bsc#1093289](https://bugzilla.suse.com/show_bug.cgi?id=1093289)
* [jsc#DOCTEAM-96](https://jira.suse.com/browse/DOCTEAM-96)

### PR creator: Which product versions do the changes apply to?

When opening a PR, check all versions of the documentation that your PR applies to.

- SLE 15/openSUSE Leap 15.x
  - [x] SLE 15 next/openSUSE Leap next *(current `main`, no backport necessary)*
  - [x] SLE 15 SP6/openSUSE Leap 15.6
  - [x] SLE 15 SP5/openSUSE Leap 15.5
  - [x] SLE 15 SP4/openSUSE Leap 15.4
  - [x] SLE 15 SP3/openSUSE Leap 15.3
  - [ ] SLE 15 SP2/openSUSE Leap 15.2

- SLE 12
  - [ ] SLE 12 SP5

### PR reviewer only: Have all backports been applied?

The doc team member merging your PR will take care of backporting to older documents.
When opening a PR, do *not* set the following check box.

- [ ] all necessary backports are done

### CODE REVIEW

**Review by taroth21 — CHANGES_REQUESTED** [review]
@sounix000 : Thanks a lot! I have read through it and it is very comprehensive indeed but written and structured very well from my POV!

Let's wait for the review by the SME you contacted and after integrating the technical feedback please hand it over to @dariavladykina for a language and style review.  I only have two minor nitpicks:

- Even though our XML schema not explicitly requires it, perhaps generally add titles to the admonitions (warning, important, tip etc.). This it at least good practice as suggested by an American norm, the ANSI Z535.6 standard.
- You used a lot of examples, which is great! For even better overview (and to make them even more findable) I would suggest to tag them as `<example>` explicitly (like you did in https://github.com/SUSE/doc-sle/blob/3519961813a046e8d29629421326577cbb611785/xml/tuning_tuned.xml#L615C9-L616C62) unless they are really short.

File: xml/product-entities.ent
Comment (taroth21) [inline xml/product-entities.ent:59]: That works fine in the URL :)

File: xml/tuning_tuned.xml
Comment (taroth21) [inline xml/tuning_tuned.xml:224]:
```suggestion
      <title>Starting TuneD</title>
```

Comment (taroth21) [reply xml/tuning_tuned.xml:224]: You can also keep 'Running' here - but in the context of the preceding and following section about enabling and disabling services I'd rather expect a section called 'Starting $SERVICE'

**Comment by sounix000** [comment]
> @sounix000 : Thanks a lot! I have read through it and it is very comprehensive indeed but written and structured very well from my POV!
> 
> Let's wait for the review by the SME you contacted and after integrating the technical feedback please hand it over to @dariavladykina for a language and style review. I only have two minor nitpicks:
> 
> * Even though our XML schema not explicitly requires it, perhaps generally add titles to the admonitions (warning, important, tip etc.). This it at least good practice as suggested by an American norm, the ANSI Z535.6 standard.
> * You used a lot of examples, which is great! For even better overview (and to make them even more findable) I would suggest to tag them as `<example>` explicitly (like you did in https://github.com/SUSE/doc-sle/blob/3519961813a046e8d29629421326577cbb611785/xml/tuning_tuned.xml#L615C9-L616C62) unless they are really short.

@taroth21 I have pushed the changes suggested by you, and some more. I will now work on Soren's technical feedback.

File: xml/tuning_tuned.xml
Comment (tomschr) [inline xml/tuning_tuned.xml:723]: @sounix000 That empty filename causes the layout issue.

File: xml/tuning_tuned.xml
Comment (tomschr) [inline xml/tuning_tuned.xml:746]: Don't indent text inside `<screen>`. Always start in column 1.

**Review by tomschr — CHANGES_REQUESTED** [review]
Just a minor issue that I found by accident. :wink:

File: xml/tuning_tuned.xml
Comment (tomschr) [inline xml/tuning_tuned.xml:40]: Typo:

```suggestion
      System administrators can apply predefined or customized tuning <emphasis>profiles</emphasis>
```

File: xml/tuning_tuned.xml
Comment (tomschr) [inline xml/tuning_tuned.xml:74]: I'd recommend to use the respective entity:

```suggestion
                A &tuned; systemd service that manages the daemon process, such
```

File: xml/tuning_tuned.xml
Comment (tomschr) [inline xml/tuning_tuned.xml:114]: Maybe consider to use `emphasis` around the `tunedapp` as well? I think it's not a general plug-in, but a tunedapp plug-in, isn't it?

```suggestion
            A <emphasis>&tunedapp; plug-in</emphasis> is an implementation of tuning logic for
```

File: xml/tuning_tuned.xml
Comment (tomschr) [inline xml/tuning_tuned.xml:184]: Use the `optional` value on `step` to make it consistent with the layout:

```suggestion
        <step performance="optional">
          <para>
            To know more, read the <emphasis>man pages</emphasis>
```

File: xml/tuning_tuned.xml
Comment (tomschr) [inline xml/tuning_tuned.xml:238]: If you are already in a procedure and have additional steps inside another step, use `<substeps>`:

```suggestion
          <substeps>
            <step>
```

Add the necessary end tags. :wink:

File: xml/tuning_tuned.xml
Comment (tomschr) [inline xml/tuning_tuned.xml:255]:
```suggestion
            </step>
            <step>
```

File: xml/tuning_tuned.xml
Comment (tomschr) [inline xml/tuning_tuned.xml:269]:
```suggestion
            </step>
          </substeps>
```

File: xml/tuning_tuned.xml
Comment (tomschr) [inline xml/tuning_tuned.xml:240]: "Lifecycle"? I'm not sure what you mean by that. Sounds a bit odd for a small daemon to have a lifecycle as I would associate something different.

Systemd have the term "states".

File: xml/tuning_tuned.xml
Comment (tomschr) [inline xml/tuning_tuned.xml:317]: These two paragraph could be marked up as `stepalternatives`:

```suggestion
        <step>
          <para>
            To stop the &tuned; daemon from being automatically activated during the next boot, you have two options:
          </para>
          <stepalternatives>
            <step>
              <para>disable the associated systemd service only:</para>
              <screen>&prompt.sudo;<command>systemctl disable tuned</command></screen>
            </step>
            <step>
              <para>
                simultaneously disable the &tuned; systemd service and
                stop the &tuned; daemon immediately:
               </para>
               <screen>&prompt.sudo;<command>systemctl disable --now tuned</command></screen>
            </step>
          </stepalternatives>
        </step>
```

File: xml/tuning_tuned.xml
Comment (tomschr) [inline xml/tuning_tuned.xml:278]: That's should actually not a separate step, but a `stepalternative` as shown below.

File: xml/tuning_tuned.xml
Comment (tomschr) [inline xml/tuning_tuned.xml:359]: I think repeating the profile name is kind of unnecessary as it's already printed in the `<term>`. You could shorten it like this:

```suggestion
              Designed for systems where CPU
              resources need to be partitioned for specific tasks or applications. It configures
```

If you want to go this path, make the other first sentences in this variablelist consistent.

File: xml/tuning_tuned.xml
Comment (tomschr) [inline xml/tuning_tuned.xml:489]: In my humble opinion, explaining what options you need is a bit unnecessary and repetitive as you show it in the screen already. I think it's enough to shorten the sentence like this:

```suggestion
        If you want to activate a profile, pass the profile name to the <command>tuned</command> command:
```

File: xml/tuning_tuned.xml
Comment (tomschr) [inline xml/tuning_tuned.xml:648]: Don't indent this content of the screen. :wink:

File: xml/tuning_tuned.xml
Comment (tomschr) [inline xml/tuning_tuned.xml:680]: Same here, no indentation.

File: xml/tuning_tuned.xml
Comment (tomschr) [inline xml/tuning_tuned.xml:661]: Not sure about this:

```suggestion
                by adding it to the <literal>[script]</literal> section:
```

**Review by tomschr — CHANGES_REQUESTED** [review]
@sounix000 I've found some minor issues and provided also some suggestions. Feel free to use them or discard them.

I'm not finished yet, but wanted to provide some ideas.

File: xml/tuning_tuned.xml
Comment (tomschr) [inline xml/tuning_tuned.xml:2517]: That's certainly a good tip, but I think there is a purpose missing: why should a customer do that? When do they have to do it? Maybe to help support? To review the changelog? To find security fixes?

For two latter use cases it would be better to use `rpm -q --changelog tuned` to view the changelog including any CVS security fixes. Looking only on GitHub can be misleading as patches can be applied in the SUSE RPM package, but not yet in GitHub.

File: xml/tuning_tuned.xml
Comment (tomschr) [inline xml/tuning_tuned.xml:1048]: Maybe make sure they are different values?

```suggestion
<replaceable>PARAMETER1</replaceable>=<replaceable>VALUE1</replaceable>
<replaceable>PARAMETER2</replaceable>=<replaceable>VALUE2</replaceable>
```

**Review by tomschr — COMMENTED** [review]
@sounix000 Found two minor things.

File: xml/tuning_tuned.xml
Comment (sounix000) [reply xml/tuning_tuned.xml:1048]: Good suggestion. Added an underscore for readability.

File: xml/tuning_tuned.xml
Comment (sounix000) [reply xml/tuning_tuned.xml:2517]: TuneD profiles and use of plug-ins are highly customizable, to suit many use cases. However, the standard stuff installed with the official package is very limited. I have given links to several GitHub resources earlier in the article, so that customers can get ideas and inspiraton for what's possible, and they can experiment a bit. 
However, the version of TuneD shipped with SLES15SP6 is v2.10.0, which is 5-6 years behind the latest version. If a customer, by mistake, takes inspiration from the latest branches or tags in the GitHub repository, instead of the branch/tag that corresponds to the installed version of TuneD, then that can be disastrous. That's because later versions of TuneD has features or capabilities that makes certain customizations of profiles and plug-ins technically valid, while the older versions such as v2.10.0 doesn't have those fancy bells and whistles. I wrote the tip so that customers don't make this mistake of taking customization ideas from the latest branches/tags, and try to make them work with older installed version of TuneD.
I hope I was able to explain my intent. I would be happy to hop on a call with you and discuss this further, if required.

**Review by taroth21 — APPROVED** [review]
As soon as the feedback by Sören and Daria are in, I'm fine with this. Many thanks!

**Review by dariavladykina — COMMENTED** [review]
Hi Souvik, please see my suggestions here. Thanks!

**Comment by sounix000** [comment]
> Hi Souvik, please see my suggestions here. Thanks!

@dariavladykina I have done the changes based on your suggestion. PTAL.

**Review by tomschr — APPROVED** [review]
Thanks @sounix000, looks great! :+1:

File: xml/tuning_tuned.xml
Comment (dariavladykina) [inline xml/tuning_tuned.xml:1819]:
```suggestion
                  <parameter>net.core.busy_read</parameter>: Number of busy loops for
```

**Review by dariavladykina — APPROVED** [review]
LGTM, just a minor follow-up. Thanks!

**Pull request merged**
